File: src/apple/Packages/MapleUI/Sources/MapleUI/MoleculesL2/MuiVisionRow.swift
// MuiVisionRow.swift — Maple UI Molecules-L2 (unified-component-catalog.md
// §3). Classification result chips, built from Chip Row. A thin,
// read-oriented wrapper: the underlying Chip Row still runs in `.select`
// mode (so a tag can be focused/highlighted), but this molecule doesn't
// surface a two-way selection — vision labels are model output, not a
// filter the caller needs to persist.

import SwiftUI

public struct MuiVisionRow: View {
    public let labels: [MuiChip]

    public init(labels: [MuiChip]) {
        self.labels = labels
    }

    public var body: some View {
        MuiChipRow(chips: labels, mode: .select, selectedId: .constant(nil))
            .accessibilityLabel("Vision labels")
    }
}

#Preview("MuiVisionRow") {
    MuiVisionRow(labels: [
        MuiChip(id: "1", label: "Mountain"),
        MuiChip(id: "2", label: "Snow"),
        MuiChip(id: "3", label: "Sunset"),
    ])
    .padding()
    .background(MuiTokens.bg)
}

File: src/windows/Maple.WinUI/MapleUI/MoleculesL2/MuiVisionRow.cs
using System.Collections.Generic;
using Microsoft.UI.Xaml;
using Microsoft.UI.Xaml.Controls;

namespace Maple.UI
{
    /// <summary>
    /// Maple.UI Vision Row molecule (unified-component-catalog.md §3,
    /// "Vision Row" row: "Classification result chips", built from Chip
    /// Row) — a thin, read-oriented wrapper: the underlying
    /// <see cref="MuiChipRow"/> still runs in Select mode (so a label can
    /// be focused/highlighted), but this molecule doesn't surface a
    /// two-way SelectedId — vision labels are model output, not a filter
    /// the caller needs to persist, matching
    /// `mui-vision-row.component.ts`'s own reasoning.
    /// </summary>
    public sealed class MuiVisionRow : ContentControl
    {
        public static readonly DependencyProperty LabelsProperty =
            DependencyProperty.Register(nameof(Labels), typeof(IReadOnlyList<MuiChip>), typeof(MuiVisionRow),
                new PropertyMetadata(null, (d, e) => ((MuiVisionRow)d)._chipRow.Chips = (IReadOnlyList<MuiChip>)e.NewValue));

        public IReadOnlyList<MuiChip>? Labels
        {
            get => (IReadOnlyList<MuiChip>?)GetValue(LabelsProperty);
            set => SetValue(LabelsProperty, value);
        }

        private readonly MuiChipRow _chipRow = new() { Mode = MuiChipRowMode.Select };

        public MuiVisionRow()
        {
            Content = _chipRow;
            IsTabStop = false;
        }
    }
}
